### PR TITLE
Add auto generated name for function

### PR DESCRIPTION
**Pull request by rJankowski93**
**Description**

Changes proposed in this pull request:

- add generated name for function
- replace custom var with string
- fix test

**Related issue(s)**

#613

### CODE REVIEW

File: cmd/kyma/init/function/function.go
Comment (clebs) [inline cmd/kyma/init/function/function.go:42]: It would be better to only generate a name if the user is not providing the flag because random generation is costly.

Comment (pPrecel) [reply cmd/kyma/init/function/function.go:42]: This PR is still `Work In Progress` ;) We know that, but thanks for your suggestions

Comment (clebs) [reply cmd/kyma/init/function/function.go]: I see, in that case please create `Draft` pull requests so that reviewers are not being notified.

File: cmd/kyma/init/function/opts.go
Comment (clebs) [inline cmd/kyma/init/function/opts.go:44]: Seeding the RNG is only necessary if you want different pseudo-random sequences in each run, which is probably not needed here.

If you still think this is necessary though, I would expect the seeding to happen inside `GenerateName`. The consumer having to do it, leaks implementation details of `GenerateName`.

Comment (rJankowski93) [reply cmd/kyma/init/function/opts.go]: I think this line is necessary, so I moved this to implementation GenerateName

Comment (clebs) [reply cmd/kyma/init/function/opts.go]: It that case, this PR can be merged after https://github.com/kyma-incubator/hydroform/pull/217 and updating the `go.mod` file.

Comment (rJankowski93) [reply cmd/kyma/init/function/opts.go]: Yes, of course, but I'm waiting for your review because if you will have another comments I will have to create new PR to hydroform.

Comment (clebs) [reply cmd/kyma/init/function/opts.go]: AH I see, I do not have any further comments. As I said in the previous comment, as soon as the other PR is merged and the go.mod file updated here, this can be merged.

Comment (rJankowski93) [reply cmd/kyma/init/function/opts.go]: Done

**Review by clebs — APPROVED** [review]

**Review by kazydek — APPROVED** [review]

**Pull request merged**
